### ISSUE TITLE
Neptune Logger Proxies

### ISSUE THREAD

**Issue by cemde** [PyTorchLightning/pytorch-lightning#8704]
## 🚀 Feature
It would be nice to use the `NeptuneLogger` with proxies as Neptune Client does support this.

### Motivation
Many compute clusters have proxies.

### Pitch

It would be nice to initialise `NeptuneLogger`like this:

```python
logger = NeptuneLogger(proxies={'http':'a', 'https':'b'})
```

The class has a method with first lines:

``` python
    def _create_or_get_experiment(self):
        if self.offline_mode:
            project = neptune.Session(backend=neptune.OfflineBackend()).get_project('dry-run/project')
        else:
            session = neptune.Session.with_default_backend(api_token=self.api_key)
```

One could easily augment

``` python
            session = neptune.Session.with_default_backend(api_token=self.api_key, proxies=self.proxies)
```
after copying `proxies` into `self.proxies`.

See: https://docs-legacy.neptune.ai/api-reference/neptune/index.html#neptune.Session

I would be happy to give it a try.

### Alternatives

N/A

### Additional context

@kamil-kaczmarek Is this impacted by the #6867 issue?

**Comment by kamil-kaczmarek** [comment]
Hi @cemde ,

I can confirm that we work on the improved `NeptuneLogger` -> you will have an option to specify proxies for `NeptuneLogger`.

**Comment by cemde** [comment]
Thanks for the reply and the upcoming improvement. I assume it will be available from v1.5. Until then I will just use my fork.
